### PR TITLE
Clean up testing environment for rzansel with XL.

### LINKED ISSUE #830: XL-build fails to throw floating point exception
The `dsxx_fpe_trap_2` unit test is failing for XL Release builds. 

```console
>>> Running:   
>>>          /scratch/users/kellyt/cdash/darwin/draco/Nightly_xlc++-power9xl-develop/Release/build/src/ds++/test/do_exception
>>>          2

- fpe_trap: This platform is supported.

- Case two: trying to evaluate sqrt(-1.0)...
  result = nan
Test Passes: Application ran to completion

Checking the generated output files...

Test Passes: Case 2 -> sqrt(-1.0) test case
Test Fails: Failed to catch SIGFPE (Invalid floating point operation)
```

I think we need to tell XL to to look for these types of floating-point errors so its behavior matches other compilers.  Trying:

* `-qflttrap` [Ref](https://www.ibm.com/support/knowledgecenter/SSGH2K_13.1.0/com.ibm.xlc131.aix.doc/compiler_ref/opt_flttrap.html)

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* When building optimized builds with IBM XL, the quadrature unit tests and `dsxx_fpe_trap_2` fail.  Some tests launched with `jsrun` would report warnings.

### Purpose of Pull Request

* Fixes #830 
* [Fixes Redmine 1949](https://rtt.lanl.gov/redmine/issues/1949)
* Might fix test failures reported by XL-based regressions on Darwin.

### Description of changes

**Changes for test registration:**

+ For _rzansel_, switch from using `jsrun` to `lrun`.  This fixes an issue where  `jsrun` was complaining about the requested number of threads being greater than the 4 hardware threads provided by each power9 core.
+ For rzansel, update the `MPIEXEC_PREFLAGS` so they work with `lrun`.
  - `MPIEXEC_PREFLAGS="--pack --threads=1 --bind=off -v"`
  - `MPIEXEC_OMP_PREFLAGS="--pack -c $ENV{OMP_NUM_THREADS} --threads=$ENV{OMP_NUM_THREADS} --bind=off -v"`
  - `MPIEXEC_NUMPROC_FLAG="-n"`
  - `MPIRUN=lrun`

**Changes to XL compiler flags:**

+ Only use `-g` for _Debug_ builds.
+ Add `-qflttrap` for all builds to mimic the default floating point exception behavior used by Intel, Gnu, Microsoft, and LLVM.
+ Reduce the _Release_ build optimization to `-O2 -qmaxmem=-1`. Enabling `-O3` produces inaccurate results in quadrature.

**Other:**

+ Make output from `quadrature_test.cc` a bit more verbose.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [ ] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by KineticTheory** [comment]
@jhchang-lanl @alexrlongne I would like your review of these changes for rzansel.  It will be good for you to be aware of the compile and lrun options that I am picking for ATS-2.  I am still iterating on these options and trying to find good choices for our codes.  The drop to `-O2` for XL might have a performance impact, but right now I can't get the tests to pass w/o it.

If we get XL + cuda working, we will need extra `lrun` options to specify the number of GPUs for each resource set.  We might also need a way to prevent multiple unit tests from trying to use the same GPU at the same time.  But this addition can wait until we have a failure mode to test against.

**Comment by jhchang-lanl** [comment]
@KineticTheory I'm good with this. Do also need extra lrun for gcc builds?

File: config/unix-xl.cmake [hunk 1]
```diff
@@ -69,11 +69,11 @@ if( NOT CXX_FLAGS_INITIALIZED )
...
 
-  # 2019-04-03 IBM support asks that we not use '-qcheck' due to compiler issues.
-  set( CMAKE_C_FLAGS_DEBUG          "-O0 -qsmp=omp:noopt -qfullpath -DDEBUG")
+  set( CMAKE_C_FLAGS_DEBUG          "-g -O0 -qsmp=omp:noopt -qfullpath -DDEBUG")
   set( CMAKE_C_FLAGS_RELWITHDEBINFO
-    "-O3 -qhot=novector -qsmp=omp -qstrict=nans:operationprecision" )
-  set( CMAKE_C_FLAGS_RELEASE        "${CMAKE_C_FLAGS_RELWITHDEBINFO} -DNDEBUG" )
+    "-g -O2 -qsmp=omp -qstrict=nans:operationprecision -qmaxmem=-1" )
+  set( CMAKE_C_FLAGS_RELEASE
+    "-O2 -qsmp=omp -qstrict=nans:operationprecision -qmaxmem=-1 -DNDEBUG" )
```
Comment (attom) [inline config/unix-xl.cmake:76]: Is -O3 bad for xl?

**Comment by KineticTheory** [comment]
> @KineticTheory I'm good with this. Do you also need extra lrun for gcc builds?

The switch from `jsrun` to `lrun` for test registration is for all compilers.  This change is based, in part, on recommendations from @zerr. It eliminates run-time warnings.  I also added the `-v` option that will print the full underlying `jsrun` command, so we can go back to `jsrun` if needed.  We still cannot pack a node when running tests.

**Comment by alexrlongne** [comment]
LGTM. Have you seen a performance hit or build time hit when using "-g" in release builds?

**Comment by KineticTheory** [comment]
> LGTM. Have you seen a performance hit or build time hit when using "-g" in release builds?

I haven't measured the performance cost of `-g`.  We should probably do this study.

I removed it to be consistent with our other builds.  The main cost will be related to library size on disk, load times and IPO.

**Pull request merged**